### PR TITLE
Add option to bump baseline of rjrinit plugin default to true

### PR DESCRIPTION
**Pull request by jtnord**
https://github.com/jenkinsci/jenkins-test-harness/pull/780#discussion_r1627958793

to get an incremental build that can then be used in the bom
https://github.com/jenkinsci/bom/pull/3284

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java [hunk 1]
```diff
@@ -444,6 +446,16 @@ public RealJenkinsRule includeTestClasspathPlugins(boolean includeTestClasspathP
         return this;
     }
 
+    /**
+     * Updates the Jenkins-Version defined by the {@code RealJenkinsRuleInit} plugin to match the version of Jenkins that will be run.
+     * The intended use case of this is to prevent any detached plugins being dragged in when they would not normally be.
+     * @param updateRealJenkinsRuleInitPluginBaseline {@code false} to skip updating the plugin, {@code true} (the default) to update the baseline.
+     */
+    public RealJenkinsRule updateRealJenkinsRuleInitPluginBaseline(boolean updateRealJenkinsRuleInitPluginBaseline) {
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:454]: Or just do it unconditionally? If there are no known problems, why even offer the option to _not_ do it?

**Review by jglick — COMMENTED** [review]
Prefer this over #780 if it works out.

File: src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java (hunk 1 above)
Comment (timja) [reply src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:454]: https://github.com/jenkinsci/jenkins-test-harness/pull/780#discussion_r1627958793

**Comment by jtnord** [comment]
closing in favor of #782

**Pull request closed without merging**